### PR TITLE
Palette (builds on top of #248)

### PR DESCRIPTION
**Pull request by mateon1**
Merge after #248

### CODE REVIEW

File: src/bg_palette_buffer.c [hunk 1]
```diff
@@ -3,9 +3,9 @@
 #include "cpu.h"
 
 #define BG_PALETTE_BUFFER_SIZE 512
-#define BG_PALETTE_BUFFER_CHUNK_SIZE 16
+#define BG_PALETTE_ROW_SIZE 16
 
-static EWRAM_DATA bool8 sBGPaletteUsed[BG_PALETTE_BUFFER_SIZE / BG_PALETTE_BUFFER_CHUNK_SIZE] = {0};
+static EWRAM_DATA bool8 sBGPaletteRowDirty[BG_PALETTE_BUFFER_SIZE / BG_PALETTE_ROW_SIZE] = {0};
```
Comment (DizzyEggg) [inline src/bg_palette_buffer.c:8]: What does `Dirty` mean here?

Comment (mateon1) [reply src/bg_palette_buffer.c:8]: It's a standard term for something being changed but not yet flushed or commited. Thus dirty caches, or a dirty git index.

Comment (DizzyEggg) [reply src/bg_palette_buffer.c:8]: I see, thanks

**Comment by mateon1** [comment]
I'll be AFK for a few hours, but should be ready to merge in current state (after the other PR ofc)

**Review by AnonymousRandomPerson — APPROVED** [review]

**Pull request merged**